File: problems/448. Find All Numbers Disappeared in an Array/solution.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        result = []
        for i in range(len(nums)):
            index = abs(nums[i]) - 1
            nums[index] = -1 * abs(nums[index])

        for j in range(len(nums)):
            if nums[j] > 0:
                result.append(j+1)
        return result

File: problems/448. Find All Numbers Disappeared in an Array/solution.cpp
class Solution
{
public:
    vector<int> findDisappearedNumbers(vector<int> &nums)
    {
        vector<int> result;
        for (int i = 0; i < nums.size(); i++)
        {
            int k = abs(nums[i]) - 1;
            if (nums[k] > 0)
                nums[k] *= -1;
        }

        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] > 0)
                result.push_back(i + 1);
        }

        return result;
    }
};